### DISCUSSION TITLE
using-cancellationtokens-in-asp-net-core-mvc-controllers/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#86] [Announcements]
# using-cancellationtokens-in-asp-net-core-mvc-controllers/

In this post I'll show how you can use CancellationTokens in your MVC action methods to stop execution when a user cancels a request from their browser

https://andrewlock.net/using-cancellationtokens-in-asp-net-core-mvc-controllers/

**Comment by sirajhussain** [comment]
Very informative post with deep knowledge.  Thank you for this article, please keep posting.

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at August 22 2017, 04:10, in reply to [Paul H](https://disqus.com/by/disqus_9VS90SRLfx/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/86#discussioncomment-6042843)</em>

---

<p>Haha, you got me;) Such a minor change but so much easier to read!</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at August 30 2017, 06:35, in reply to [Mate Antičević](https://disqus.com/by/mateantievi/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/86#discussioncomment-6042850)</em>

---

<p>Which part didn't work for you? What behaviour did you see?</p>

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at August 30 2017, 10:19, in reply to [Mate Antičević](https://disqus.com/by/mateantievi/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/86#discussioncomment-6042853)</em>

---

<p>Interesting, I'll look into it</p>

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at August 30 2017, 10:20, in reply to [Andrzej Pauli](https://disqus.com/by/andrzejpauli/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/86#discussioncomment-6042855)</em>

---

<p>Definitely works in Edge, that's what I tested it in for this post :)</p>

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at August 31 2017, 08:40, in reply to [Mate Antičević](https://disqus.com/by/mateantievi/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/86#discussioncomment-6042853)</em>

---

<p>You're completely right - this doesn't work with either IIS or IIS Express. Went digging through issues and it's a known problem with the AspNetCoreModule that forwards requests to Kestrel: <a href="https://github.com/aspnet/AspNetCoreModule/issues/38" rel="nofollow noopener" title="https://github.com/aspnet/AspNetCoreModule/issues/38">https://github.com/aspnet/AspNetCoreModule/issues/38</a>. <br>Additional/related info at <a href="https://github.com/aspnet/Mvc/issues/5239" rel="nofollow noopener" title="https://github.com/aspnet/Mvc/issues/5239">https://github.com/aspnet/Mvc/issues/5239</a> and <a href="https://github.com/aspnet/KestrelHttpServer/issues/1139" rel="nofollow noopener" title="https://github.com/aspnet/KestrelHttpServer/issues/1139">https://github.com/aspnet/KestrelHttpServer/issues/1139</a></p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at September 26 2017, 06:29, in reply to [Muhammad Rehan Saeed](https://disqus.com/by/RehanSaeedUK/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/86#discussioncomment-6042869)</em>

---

<p>Cool, thanks :) I'll update the post to mention that</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at December 05 2019, 03:57, in reply to [Stefan Minchev](https://disqus.com/by/stefan_minchev/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/86#discussioncomment-6042884)</em>

---

<p>Hi, can't remember exactly, but at least version 2.0+, and yes, Web APIs and MVC Controllers are exactly the same thing in ASP.NET Core so you can use them in both! The naming essentially just reflects them being used for slightly different things, and having slightly different conventions 🙂</p>

[a comment by an automated account is not shown]

**Comment by code-bode** [comment]
Thanks for putting this amazing article. II need to timeout after 2 minute if `slowtest` take more time to return the data. I tried to follow some of your example but can't figure out how to do it. I have seen .net core 8 has introduce requestTimeOut attribute but that only works if i am calling another service under `slowtest` call.

**Reply by denisbredikhin** [reply]
Hello Andrew,

you say that we can use while (!IsCancellationRequested) instead of ThrowIfCancellationRequested, but Stephen Cleary here says that it is a bad practice:
https://blog.stephencleary.com/2022/03/cancellation-4-polling.html

**Reply by denisbredikhin** [reply]
Sorry wrong thread....

**Comment by denisbredikhin** [comment]
Hello Andrew,

you say that we can use while (!IsCancellationRequested) instead of ThrowIfCancellationRequested, but Stephen Cleary here says that it is a bad practice:
https://blog.stephencleary.com/2022/03/cancellation-4-polling.html
